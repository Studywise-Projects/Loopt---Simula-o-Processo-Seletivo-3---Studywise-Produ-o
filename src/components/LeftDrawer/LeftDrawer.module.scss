@import '@/styles/_global.scss';

.swipeableDrawer {
  padding: 32px;
}

.iconButton {
  color: $white;
  padding: 0;
}

.link {
  text-decoration: none;
}

.link:active {
  text-decoration: none;
  font-weight: 700;
}

.listItem {
  padding: 8px 0;
}

.listIcon {
  color: $black;
}

.listIcon:active {
  font-weight: 700;
}

.listText {
  color: $black;
}

.listText:active {
  font-weight: 700;
}