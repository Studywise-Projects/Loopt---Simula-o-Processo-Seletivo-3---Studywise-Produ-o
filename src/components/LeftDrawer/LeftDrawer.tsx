import React from 'react';
import {
  IconButton,
  List,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  SwipeableDrawer,
} from '@mui/material';
import { useState } from 'react';
import MenuIcon from '@mui/icons-material/Menu';
import HomeIcon from '@mui/icons-material/Home';
import StarIcon from '@mui/icons-material/Star';
import PeopleIcon from '@mui/icons-material/People';
import Logo from '../Logo/Logo';
import styles from './LeftDrawer.module.scss';
import SelectAutoComplete from '../SelectAutoComplete/SelectAutoComplete';
import Link from 'next/link';

interface ILeftDrawer {
  optionsSelect: any;
  valueSelect: Object | null;
  handleChangeSelect: any;
}

function LeftDrawer({
  optionsSelect,
  valueSelect,
  handleChangeSelect,
}: ILeftDrawer) {
  const [state, setState] = useState(false);

  const toggleDrawer =
    (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
      if (
        event &&
        event.type === 'keydown' &&
        ((event as React.KeyboardEvent).key === 'Tab' ||
          (event as React.KeyboardEvent).key === 'Shift')
      ) {
        return;
      }

      setState(open);
    };

  return (
    <>
      <IconButton onClick={toggleDrawer(true)} className={styles.iconButton}>
        <MenuIcon sx={{ fontSize: '32px' }} />
      </IconButton>
      <SwipeableDrawer
        anchor='left'
        open={state}
        onClose={toggleDrawer(false)}
        onOpen={toggleDrawer(true)}
        className={styles.swipeableDrawer}
        sx={{
          '& .MuiPaper-root': {
            gap: '16px',
            padding: '32px',
          },
          '& .MuiDrawer-paper': {
            boxSizing: 'border-box',
            paddingTop: '16px',
            width: '300px',
          },
        }}
      >
        <Logo variant='horizontal' />
        <SelectAutoComplete
          options={optionsSelect}
          label='Vaga'
          value={valueSelect}
          handleChange={handleChangeSelect}
          isAuxSelect={true}
        />
        <List>
          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <HomeIcon />
              </ListItemIcon>
              <ListItemText primary='Home' className={styles.listText} />
            </ListItemButton>
          </Link>

          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <StarIcon />
              </ListItemIcon>
              <ListItemText
                primary='Candidato Escolhido'
                className={styles.listText}
              />
            </ListItemButton>
          </Link>

          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <PeopleIcon />
              </ListItemIcon>
              <ListItemText
                primary='Todos os Candidatos'
                className={styles.listText}
              />
            </ListItemButton>
          </Link>
        </List>
      </SwipeableDrawer>
    </>
  );
}

export default LeftDrawer;
