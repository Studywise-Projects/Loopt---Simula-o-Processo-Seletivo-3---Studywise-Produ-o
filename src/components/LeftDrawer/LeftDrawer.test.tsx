import { render } from '@testing-library/react';
import LeftDrawer from './LeftDrawer';

jest.mock('next/router', () => require('next-router-mock'));

describe('JobDetailsCard', () => {
  it('Should match the snapshot', () => {
    const { getByLabelText } = render(
      <LeftDrawer
        optionsSelect={[{ id: 1, label: 'Desenvolvedor Front-End' }]}
        valueSelect={null}
        handleChangeSelect={undefined}
      />,
    );

    const element = getByLabelText('Vaga');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByLabelText } = render(
      <LeftDrawer
        optionsSelect={[{ id: 1, label: 'Desenvolvedor Front-End' }]}
        valueSelect={null}
        handleChangeSelect={undefined}
      />,
    );

    const element = getByLabelText('Vaga');
    expect(element).toBeInTheDocument();
  });
});
