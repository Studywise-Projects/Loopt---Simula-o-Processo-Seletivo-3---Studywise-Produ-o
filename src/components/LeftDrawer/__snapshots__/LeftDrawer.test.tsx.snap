// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftDrawer Should match the snapshot 1`] = `ShallowWrapper {}`;
